client/extensions/woocommerce/app/order/order-details/table.js

```diff
@@ -74,24 +88,69 @@ class OrderDetailsTable extends Component {
 		);
 	};
 
-	renderOrderItems = item => {
-		const { order, site } = this.props;
+	onChange = event => {
+		const { order } = this.props;
+		// Name is `quantity-x`, where x is the ID of the item
+		const id = parseInt( event.target.name.split( '-' )[ 1 ] );
+		const item = find( order.line_items, { id } );
+		if ( ! item ) {
+			return;
+		}
+		const index = findIndex( order.line_items, { id } );
+		// A zero quantity does strange things with the price, so we'll force 1
+		const quantity = parseInt( event.target.value ) || 1;
+		const subtotal = getOrderItemCost( order, id ) * quantity;
+		const total = subtotal;
+		const newItem = { ...item, quantity, subtotal, total };
+		this.props.onChange( { [ index ]: newItem } );
+	};
+
+	renderQuantity = item => {
+		const { isEditing } = this.props;
+		if ( isEditing ) {
+			return (
+				<FormTextInput
+					type="number"
+					min={ 1 }
+					name={ `quantity-${ item.id }` }
+					onChange={ this.onChange }
+					value={ item.quantity }
+				/>
+			);
+		}
+		return item.quantity;
+	};
+
+	renderName = item => {
```

Comment:
Sorry - another nit - could we name this `renderItemName` to disambiguate for maintainers from the customer's name?

Reply:
There is no customer info in this component, I think it's OK to leave as-is.